outscale/config.go

```diff
@@ -47,13 +48,18 @@ func (c *Config) Client() (*OutscaleClient, error) {
 			Proxy:           http.ProxyFromEnvironment,
 		},
 	}
-	skipClient.Transport = logging.NewTransport("Outscale", skipClient.Transport)
-	skipClient.Transport = NewTransport(c.AccessKeyID, c.SecretKeyID, c.Region, skipClient.Transport)
 
+	skipClient.Transport = logging.NewSubsystemLoggingHTTPTransport("Outscale", skipClient.Transport)
+	endpoint, isEndpointSet := c.Endpoints["api"]
+	if !isEndpointSet && c.Region == "" {
```

Comment:
You should also test that endpoint != "".

If Endpoint was a map[string]string (or a struct), you could have done:

```
endpoint := c.Endpoints["api"]
if endpoint == "" && c.Region == "" {
```